go/chat/attachment_crypt.go

```diff
@@ -10,36 +16,83 @@ type Encrypter interface {
 	// Encrypt takes a plaintext reader and returns a ciphertext reader.
 	Encrypt(plaintext io.Reader) (ciphertext io.Reader)
 
-	// Key returns the ephemeral key that was used during Encrypt.
-	Key() []byte
+	// EncryptKey returns the ephemeral key that was used during Encrypt.
+	EncryptKey() []byte
+
+	// VerifyKey returns the public portion of the signing key that
+	// can be used for signature verification.
+	VerifyKey() []byte
 }
 
 type Decrypter interface {
-	// Decrypt takes a ciphertext reader and a key and returns a
-	// plaintext reader.
-	Decrypt(ciphertext io.Reader, key []byte) (plaintext io.Reader, err error)
+	// Decrypt takes a ciphertext reader, encryption and verify keys.
+	// It returns a plaintext reader.
+	Decrypt(ciphertext io.Reader, encKey, verifyKey []byte) (plaintext io.Reader)
 }
 
-// PassThrough implements Encrypter and Decrypter.  It does no encryption
-// or decryption.
-type PassThrough struct{}
+var nonce = &[signencrypt.NonceSize]byte{
+	0x10, 0x10, 0x10, 0x10, 0x20, 0x20, 0x20, 0x20,
+	0x30, 0x30, 0x30, 0x30, 0x40, 0x40, 0x40, 0x40,
+}
```

Comment:
I like to make these constants actual ASCII strings, so that they're kind of tied to the context in which they were generated. Maybe something like `kbchatattachment`? Or alternatively a truly random 16-byte value?